Feat(trends): Add web vitals to the trends view

wmak opened the pull request:
- Adds measurements support to the trends endpoint
- Also switched to f-strings 🤷
- Use the same method we mock transaction trends to mock measurement trends as well

<img width="1388" alt="image" src="https://user-images.githubusercontent.com/4205004/106229975-923ab900-61bc-11eb-83cb-e7136b2afc90.png">

- This removes the confidence selector from trends, and defaults to only
  show what was previously known as high confidence trends
    - If necessary the confidence can still be overwritten by adding `t_test()` to the query
    - the query was purposefully left as `0<transaction.duration<15min` so that it can still exclude outliers

![image](https://user-images.githubusercontent.com/4205004/106230118-ca41fc00-61bc-11eb-835b-1592dcfe0044.png)
- This adds a new dropdown to the transaction summary so you can select which parameter to run the function on

wmak committed:
feat(trends): Adding support for web vitals to trends

wmak committed:
ref: f-strings f-yeah

wmak committed:
chore: Mock measurement trends too

wmak committed:
feat(trends): Add display parameter to the trends view

- This removes the confidence selector from trends, and defaults to only
  show what was previously known as high confidence trends
- This adds a new dropdown to the transaction summary so you can select
  which parameter to run the function on
- This allows trends to be run on web vitals as well

Zylphrex commented on src/sentry/static/sentry/app/views/performance/trends/utils.tsx:
```diff
@@ -32,54 +39,67 @@ export const DEFAULT_MAX_DURATION = '15min';
+    label: 'average',
     field: TrendFunctionField.AVG,
     alias: 'avg_range',
     chartLabel: 'avg(transaction.duration)',
```
I believe the `chartLabel` needs to be updated as well.

Zylphrex commented on src/sentry/static/sentry/app/views/performance/trends/utils.tsx:
```diff
@@ -32,54 +39,67 @@ export const DEFAULT_MAX_DURATION = '15min';
     alias: 'percentile_range',
     chartLabel: 'p50()',
     legendLabel: 'p50',
+    aggregation: 'p50',
```
I think this key needs to be added to the type definition of `TrendFunction` and if it's added as `AggregateKey`, maybe you can skip the cast from `string` in `generateTrendFunctionAsString`

Zylphrex commented on tests/sentry/api/test_event_search.py:
```diff
@@ -2589,6 +2604,20 @@ def test_average_range(self):
             ]
         ]
 
+        # Test a non duration field
```
might be a little redundant, but should we add the analogous test for all the `*_range` functions?

Zylphrex commented on src/sentry/static/sentry/app/views/performance/trends/utils.tsx:
```diff
@@ -121,12 +141,22 @@ export function getCurrentTrendFunction(location: Location): TrendFunction {
+  return generateFieldAsString({
+    kind: 'function',
+    function: [trendFunction as AggregationKey, trendParameter, undefined],
+  } as Column);
```
Is this cast to `Column` necessary here?

Zylphrex commented on src/sentry/static/sentry/app/views/performance/trends/content.tsx:
```diff
@@ -94,24 +94,16 @@ class TrendsContent extends React.Component<Props, State> {
-  handleConfidenceChange = (label: string) => {
-    const {organization, location} = this.props;
-
-    trackAnalyticsEvent({
```
Will you be adding the analytics for the column dropdown?

wmak commented on src/sentry/static/sentry/app/views/performance/trends/types.ts:
```diff
@@ -39,11 +39,19 @@ export enum TrendChangeType {
+  AVG = 'avg',
+}
+
+export enum TrendColumnField {
```
Not sure to be honest, I based this on the home page to keep these two consistent. I believe there are still times where fcp and fp can differ so we should probably keep it somewhere. Not sure if we want trends on both though. cc @k-fish since I know you likely have opinions on this

k-fish replied:
There are times where they are different, but it's extremely rare to see iirc. I think it's probably safe to remove from trends.

wmak committed:
Merge branch 'master' into feat/web-vital-trends

wmak committed:
ref: Addressing PR comments

- Switching to TrendFunction.field instead of introduceing aggregation
- Using parameter for the chartLabel instead of hardcoding it
- extra `Column` cast
- Analytics on parameter change TODO: update reload

wmak committed:
ref: More range tests

Zylphrex commented on src/sentry/static/sentry/app/views/performance/trends/changedTransactions.tsx:
```diff
@@ -392,7 +389,8 @@ function TrendsListItem(props: TrendsListItemProps) {
   const percentChangeExplanation = t(
-    'Over this period, the duration for %s has %s %s from %s to %s',
+    'Over this period, the %s for %s has %s %s from %s to %s',
+    currentTrendColumn,
     currentTrendFunction,
```
I think these two parameters need to be flipped for the sentence to make sense.

wmak committed:
Merge branch 'master' into feat/web-vital-trends

wmak committed:
fix: Flipping explanation text

wmak committed:
ref: Removing six
